unit_test/fuzzing/test_secured_message/test_spdm_encode_secured_message/spdm_encode_secured_message.c

```diff
@@ -56,14 +56,15 @@ void libspdm_test_encode_secured_message(void **State)
                              sizeof(spdm_secured_message_cipher_header_t) +
                              32; /* MCTP_MAX_RANDOM_NUMBER_COUNT */
     LIBSPDM_ASSERT(spdm_test_context->test_buffer_size > record_header_max_size);
-
-    secured_message_size = spdm_test_context->test_buffer_size - record_header_max_size;
+    secured_message_size = spdm_test_context->test_buffer_size;
+    secured_message = (void *)malloc(secured_message_size);
 
```

Comment:
why you need malloc?

Reply:
1.secured_ message requires memory to complete encode:
line 266 do assignment operation
https://github.com/DMTF/libspdm/blob/bd5cde40aad7df7d07a54df5b11664d0fb53b1ed/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L261-L266
2.secured_ message is the AUTH session, including the header
https://github.com/DMTF/libspdm/blob/bd5cde40aad7df7d07a54df5b11664d0fb53b1ed/library/spdm_secured_message_lib/libspdm_secmes_encode_decode.c#L252-L255

Reply:
1) libspdm already defined maximum size. Why not use that in stack? Why you have to use heap?

2) If APP data is not consumed, why you care? Basically, you just fuzz the App_Length.